### DISCUSSION TITLE
Windows .exe program to add text overlay on camera video

### DISCUSSION THREAD

**Discussion by eliangbb** [AxisCommunications/discussions#19] [Axis Windows SDKs] [answered]
Hi there, 
Is there a way to build Windows .exe program (in c/c++) and run .exe to add text overlay on the camera video by using Windows SDK or other SDKs? 
I built sample axoverlay APP (.eap file) by using ACAP native SDK, the APP can be added/installed/ran the on the camera, and the text overlay worked. I want to make this text overlay work from Windows .exe program instead of using APP. Is there a way to do so? 
Any help will be greatly appreciated!

**Comment by pataxis** [comment]
Hi @eliangbb , I have looked for someone that could help answer this question but it seems many are still on vacation so just wanted to give a heads up that it may take some more weeks.

**Comment by anders-axis** [comment]
The VAPIX API of the camera support configuring and customizing the burned-in overlays in the video. It should be simple to create an Windows desktop application to control this. For more information on overlay API:
https://www.axis.com/vapix-library/subjects/t10175981/section/t10007638/display

**Reply by eliangbb** [reply]
Thank you Anders. Really appreciated. I do have some concerns about the total number of lines and positions shown on the image. The attached screen below is what we are looking for about POS register transaction data. Is it possible to add multiple lines (one line by one line)? Thanks again.
![textOverlay](https://user-images.githubusercontent.com/108364188/182387750-7af2316a-9583-4e9d-8f0e-5dbc0abf4946.png)

**Reply by anders-axis** [reply]
Yes, line-breaks seems to be supported by the /axis-cgi/dynamicoverlay/dynamicoverlay.cgi API.
If you can do it in the web interface of the camera it should also be possible to do using the API.

**Reply by eliangbb** [reply]
will try. thx.

**Reply by eliangbb** [reply]
Hi Anders,
I tried the VAPIX api like this:
**_http://10.0.0.246/axis-cgi/dynamicoverlay.cgi?action=settext&camera=1&text=Test%20texttop_**
and it shows something like the screen below. But I still don't know how to add more lines. So, how to add line-breaks in the command?
![Untitled](https://user-images.githubusercontent.com/108364188/185498395-be251406-aab9-470a-af27-229ba1f4f97b.jpg)

**Reply by anders-axis** [reply]
Have you managed to create an two-line text overlay using the web-interface of the camera? What firmware version are you using on the camera?

**Reply by eliangbb** [reply]
Hi Anders, 
I tried using web-interface but it seems not what we are looking for. We need to add/insert POS transaction data (text lines) in real time from our program into the video. Our Axis device is P1455-LE 29 mm Network Camera, and the firmware version is 10.4.0  
(Note: we updated on 7/11/2022)
thx.

**Reply by anders-axis** [reply]
Yes, I understand, I just wanted to make sure the camera/firmware supported line-breaks using the web-interface.

**Reply by eliangbb** [reply]
Hi Anders,
So, does this mean there is no way to use "dynamic overlay" API to show more lines on the video at the same time in any positions? (Note: one line works fine on TOP and BOTTOM only)

**Comment by anders-axis** [comment]
I can add multiple text-overlays including multiple lines on my camera (see image) using the web interface of the camera. Please confirm if you can do this manually as well. If you can it should be possible to do it also programmatically using VAPIX API, but let's first confirm what your camera can do in the user-interface.

![image](https://user-images.githubusercontent.com/110470856/187869664-b4bb11e3-c721-48be-9c85-8b105c1ac2d0.png)

Note that I'm using the latest version of the web interface - sometimes you can chose different versions of the web interface by clicking the question-mark or menu icon at the right top corner of the web-page.

**Reply by eliangbb** [reply]
Hi Anders, 
Thank you so much. I can add more lines now by using newer web interface as you did. 
So, what kind of character (in between "Line1" and "Line2") can I put into my VAPIX command to do this programmatically? "http://10.0.0.246/axis-cgi/dynamicoverlay.cgi?action=settext&camera=1&text=Line1%0D%0ALine2" does not work. Should I use "%0D%0A" for new line? 
Thank you.

**Reply by anders-axis** [reply, marked as answer]
Hi,
If I look at the request made from the web interface it seems to use a bit richer API version with a POST payload.
![image](https://user-images.githubusercontent.com/110470856/188078657-80a72fbb-2597-4ad4-82a8-42fca2de06ce.png)

I found the corresponding API documentation here:
https://www.axis.com/vapix-library/subjects/t10175981/section/t10007638/display?section=t10007638-t10119641

It is even promoting that it supports multiple lines. :)
![image](https://user-images.githubusercontent.com/110470856/188078975-f861285c-7ecb-4b61-baba-c73d5072c1c2.png)

Feel free to check it out!

**Reply by eliangbb** [reply]
This really helps, thank you so much, Anders. Finally I got VAPIX POST-addText work with multiple lines by using your feedback. Just share with you: I used https://web.postman.co to test VAPIX and it looked great. Again, thanks.
